### PR TITLE
c++ compatibility: rename reserved C++ identifiers 'public' and 'this'

### PR DESCRIPTION
**Pull request by danielappiagyei-bc**
## Summary
Rename stream struct fields named `public` and functions parameters named `this` in `streams.h` and `rtti.h` so C++ compilers won't fail to compile them (these are [reserved keywords](https://en.cppreference.com/w/cpp/keyword) in C++). This fixes the issue called out by https://github.com/apache/nuttx/issues/10556

## Impact
`lib/libc/stream`, `segger/rtt.h`
## Testing
- Successfully compiled `nsh` app for IMXRT after I:
  - modified it to include [segger/rtt.h](https://github.com/apache/nuttx/blob/master/include/nuttx/segger/rtt.h)
  - set the following [segger Kconfigs](https://github.com/apache/nuttx/blob/master/drivers/segger/Kconfig):  `STREAM_RTT`, `SEGGER_RTT`

### CODE REVIEW

**Comment by danielappiagyei-bc** [comment]
looks like I didn't properly rebase and squash into one commit. Will fix

**Comment by xiaoxiang781216** [comment]
> looks like I didn't properly rebase and squash into one commit. Will fix

Yes, please squash you patchset:
```
git rebase --interactive HEAD~3
git push -f origin
```

**Comment by danielappiagyei-bc** [comment]
> > looks like I didn't properly rebase and squash into one commit. Will fix
> 
> Yes, please squash you patchset:
> 
> ```
> git rebase --interactive HEAD~3
> git push -f origin
> ```

Sorry for the delay, I've rebased and squash now. Thanks!

**Comment by danielappiagyei-bc** [comment]
I've fixed a few missing updates and also created a PR in the nuttx-apps repo (https://github.com/apache/nuttx-apps/pull/2047) for the files there that need to be changed. I'm not sure what the workflow is like when changes need to be applied to both repos at the same time for either PR to pass

**Comment by xiaoxiang781216** [comment]
> I've fixed a few missing updates and also created a PR in the nuttx-apps repo ([apache/nuttx-apps#2047](https://github.com/apache/nuttx-apps/pull/2047)) for the files there that need to be changed. I'm not sure what the workflow is like when changes need to be applied to both repos at the same time for either PR to pass

ci can't handle the dependence across multiple git yet, so the maintainer has to merge it after carefully reviewing.

**Comment by xiaoxiang781216** [comment]
@danielappiagyei-bc but the change from 'this' to 'self' lose?

**Comment by xiaoxiang781216** [comment]
> > @danielappiagyei-bc but the change from 'this' to 'self' lose?
> 
> I only renamed the 'this' function argument to 'self' [here](https://github.com/apache/nuttx/pull/10565/files#diff-2dd8bd975a0b4fe946a837961405b253a447eb9822e8e9a4b43359aeea2b0188L73) in `include/nuttx/streams.h` since that's the only `.h` file I found it in. It's still used in several `.c` files such as https://github.com/apache/nuttx/blob/master/drivers/segger/stream_rtt.c#L87 but I left those alone since only the `.h` files need to be compatible with C++ compilers when you `#include` them. However, I can rename these remaining usages as well if you would like

it's better to unify the naming and avoid the user copy public/this to other places.

**Review by xiaoxiang781216 — APPROVED** [review]

**Comment by danielappiagyei-bc** [comment]
Yeah thats true. It's an easy change on my end, I'll have the changes in maybe 9 hours when im free. Thanks!

File: arch/arm64/src/common/arm64_initialize.c
Comment (xiaoxiang781216) [inline arch/arm64/src/common/arm64_initialize.c:148]: change to block

File: libs/libc/stream/lib_bufferedoutstream.c
Comment (xiaoxiang781216) [inline libs/libc/stream/lib_bufferedoutstream.c:54]: remove the extra indent

File: libs/libc/stream/lib_bufferedoutstream.c
Comment (xiaoxiang781216) [inline libs/libc/stream/lib_bufferedoutstream.c:76]: ditto

File: libs/libc/stream/lib_hexdumpstream.c
Comment (xiaoxiang781216) [inline libs/libc/stream/lib_hexdumpstream.c:86]: ditto

File: libs/libc/stream/lib_stdsostream.c
Comment (xiaoxiang781216) [inline libs/libc/stream/lib_stdsostream.c:126]: why change

File: libs/libc/stream/lib_stdsostream.c
Comment (xiaoxiang781216) [inline libs/libc/stream/lib_stdsostream.c:111]: why change

File: libs/libc/stream/lib_stdsostream.c
Comment (xiaoxiang781216) [inline libs/libc/stream/lib_stdsostream.c:74]: why change

**Comment by xiaoxiang781216** [comment]
@danielappiagyei-bc please squash your change

**Comment by danielappiagyei-bc** [comment]
> @danielappiagyei-bc please squash your change

squash just the last 2 commits, correct?

File: libs/libc/stream/lib_stdsostream.c
Comment (danielappiagyei-bc) [reply libs/libc/stream/lib_stdsostream.c:74]: nxstyle complained that the line was too long so I shortened the names to be under 80 characters

File: arch/arm64/src/common/arm64_initialize.c
Comment (danielappiagyei-bc) [reply arch/arm64/src/common/arm64_initialize.c:148]: sure

File: libs/libc/stream/lib_bufferedoutstream.c
Comment (danielappiagyei-bc) [reply libs/libc/stream/lib_bufferedoutstream.c:54]: good catch, my editor added indents here for some reason

Comment (danielappiagyei-bc) [reply libs/libc/stream/lib_bufferedoutstream.c]: i've now reviewed my changes and fixed the extra indents in all files I modified, not just the ones you mentioned. good eye!

File: libs/libc/stream/lib_stdsostream.c
Comment (danielappiagyei-bc) [reply libs/libc/stream/lib_stdsostream.c:74]: I'm just now realizing that it isn't required for the `(` to be at the same indent level as the `=` so I could have kept the same name and pushed the second line to the left more:

```
FAR struct lib_stdsostream_s *stdsostream = 
                                                        (FAR struct lib_stdsostream_s *)self;
```

**Comment by xiaoxiang781216** [comment]
@danielappiagyei-bc please squash your change into one patch.

**Comment by danielappiagyei-bc** [comment]
> @danielappiagyei-bc please squash your change into one patch.

@xiaoxiang781216 done

File: libs/libc/stream/lib_stdsostream.c
Comment (danielappiagyei-bc) [reply libs/libc/stream/lib_stdsostream.c:126]: sure, consistency is good. I was worried `stream->stream` would look awkward but on second thought, it should be fine

Comment (mu578) [reply libs/libc/stream/lib_stdsostream.c:126]: stream->handle

Comment (danielappiagyei-bc) [reply libs/libc/stream/lib_stdsostream.c:126]: I like that idea, I'll get started

**Comment by danielappiagyei-bc** [comment]
@mu578 @xiaoxiang781216 
Does nuttx have any automatic reformatting tool, like clang-format or something?

File: libs/libc/stream/lib_lzfcompress.c
Comment (xiaoxiang781216) [inline libs/libc/stream/lib_lzfcompress.c:69]: need change line 86

File: include/nuttx/streams.h
Comment (danielappiagyei-bc) [inline include/nuttx/streams.h:408]: TODO: fix indent

File: libs/libc/stream/lib_lzfcompress.c
Comment (danielappiagyei-bc) [reply libs/libc/stream/lib_lzfcompress.c:69]: done

**Comment by danielappiagyei-bc** [comment]
I've now rebased into a single commit, let me know if there's any other changes required before merge please @xiaoxiang781216

**Review by xiaoxiang781216 — APPROVED** [review]

**Comment by xiaoxiang781216** [comment]
> @mu578 @xiaoxiang781216 Does nuttx have any automatic reformatting tool, like clang-format or something?

you can try:
https://github.com/apache/nuttx/blob/master/tools/uncrustify.cfg
https://github.com/apache/nuttx/blob/master/tools/detab.c
https://github.com/apache/nuttx/blob/master/tools/rmcr.c

**Comment by TimJTi** [comment]
@danielappiagyei-bc @xiaoxiang781216 
looks like the apps directory was not included when this was done. Have found 4 references to stream.public that now cause a build fail.

Perhaps the files in the app folder can be "fixed" too? I will raise a new issue for this in the nuttx-apps repo.

**Comment by TimJTi** [comment]
> > @danielappiagyei-bc @xiaoxiang781216 looks like the apps directory was not included when this was done. Have found 4 references to stream.public that now cause a build fail.
> > Perhaps the files in the app folder can be "fixed" too? I will raise a new issue for this in the nuttx-apps repo.
> 
> @TimJTi I apologize, I thought I had fixed everything in the apps repo. Yes, create a new issue please. Is CI failing too or just a specific local build?

At this stage a local build is failing - not just because of this but because of some other recent NuttX PR that means libapps.a can't be found or isn't being created!

**Comment by TimJTi** [comment]
See apps [PR2078](https://github.com/apache/nuttx-apps/issues/2078)

**Pull request merged**
